dynatrace/tests/test_dynatrace_topology.py

```diff
@@ -0,0 +1,207 @@
+        expected_topology = load_json_from_file("expected_service_topology.json")
+        self.assert_topology(expected_topology, test_topology)
+
+    def assert_topology(self, expected_topology, test_topology):
+        """
+        Sort the keys of components and relations, so we can actually match it
+        :param expected_topology: expected topology read from file
+        :param test_topology: topology gathered during test
+        :return: None
+        """
+        components, relations = sort_topology_data(test_topology)
+        expected_components, expected_relations = sort_topology_data(expected_topology)
+        self.assertEqual(components, expected_components)
+        self.assertEqual(len(relations), len(expected_relations))
+        for relation in relations:
+            self.assertIn(relation, expected_relations)
+
+    @requests_mock.Mocker()
+    def test_collect_applications(self, m):
+        """
+        Testing Dynatrace check should collect applications and also the tags properly coming from dynatrace
+        """
+        self._set_http_responses(m, apps=read_file("application_response.json"))
+
+        self.check.url = self.instance.get('url')
+
+        self.check.run()
+
+        topo_instances = topology.get_snapshot(self.check.check_id)
+        expected_topology = load_json_from_file("expected_application_topology.json")
+
+        # sort the keys of components and relations, so we match it in actual
+        self.assert_topology(expected_topology, topo_instances)
+
+    @requests_mock.Mocker()
+    def test_collect_process_groups(self, m):
+        """
+        Testing Dynatrace check should collect process-groups
+        """
+        self._set_http_responses(m, proc_groups=read_file("process-group_response.json"))
+
+        self.check.url = self.instance.get('url')
+        self.check.run()
+
+        topo_instances = topology.get_snapshot(self.check.check_id)
+        actual_topology = load_json_from_file("expected_process-group_topology.json")
+
+        # sort the keys of components and relations, so we match it in actual
+        self.assert_topology(actual_topology, topo_instances)
+
+    @requests_mock.Mocker()
+    def test_collect_relations(self, m):
+        """
+        Test to check if relations are collected properly
+        """
+        self._set_http_responses(m, hosts=read_file("host_response.json"))
+        self.check.url = self.instance.get('url')
+        self.check.run()
+
+        topo_instances = topology.get_snapshot(self.check.check_id)
+        self.assertEqual(len(topo_instances['components']), 2)
+        self.assertEqual(len(topo_instances['relations']), 5)
+
+        # since all relations are to this host itself so target id is same
+        relation = topo_instances['relations'][0]
+        self.assertEqual(relation['target_id'], 'HOST-6AAE0F78BCF2E0F4')
+        self.assertIn(relation['type'], ['isProcessOf', 'runsOn', 'isSiteOf'])
```

Comment:
we are not creating any relation for `isSiteOf`, correct ?

Reply:
removed `isSiteOf` from assert